sys/include/net/eui48.h

```diff
@@ -56,6 +56,27 @@ static inline void eui48_to_eui64(eui64_t *eui64, const eui48_t *addr)
     eui64->uint8[7] = addr->uint8[5];
 }
 
+/**
+ * @brief   Generates an EUI-48 from a 64-bit device address
+ *
+ * @warning The resulting EUI-48 is not guaranteed to be unique.
+ *
+ * @param[out] eui48    the resulting EUI-48.
+ * @param[in]  addr     a 64-bit device address
+ */
+static inline void eui64_to_eui48(eui48_t *eui48, const eui64_t *addr)
+{
+    /* Preserve vendor id */
+    eui48->uint8[0] = addr->uint8[0];
+    eui48->uint8[1] = addr->uint8[1];
+    eui48->uint8[2] = addr->uint8[2];
+
+    /* Use most volatile bits */
+    eui48->uint8[3] = addr->uint8[5];
+    eui48->uint8[4] = addr->uint8[6];
+    eui48->uint8[5] = addr->uint8[7];
```

Comment:
```suggestion
    eui48->uint8[5] = addr->uint8[7];
    eui48_set_local(eui48);
```

Reply:
And I think `eui48_clear_group()` should also be done.

Reply:
And maybe `@post` Doxygen command to document that the generated EUI48 is marked as only locally unique and unicast?

Reply:
Hm, `eui48_to_eui64()` doesn't modify those bits either.
In the end it would be used as a normal 'raw' EUI-48, there the users already take care of setting / clearing those bits if needed.

Reply:
Indeed, the group bit should already be cleared. But IMO a globally unique EUI-64 should still be turned into a locally unique EUI-48 for the reasons mentioned in the discussion below.